### PR TITLE
Version 0.17.0

### PR DESCRIPTION
**Pull request by Kludex**
Changes: [0.16.0..master](https://github.com/encode/starlette/compare/0.16.0...master)
GitHub Draft release: [here](https://github.com/encode/starlette/releases/tag/untagged-8aef014e611c6bd03153)

### Missing
- [x] https://github.com/encode/starlette/pull/1262
- [ ] Anything else?

### CODE REVIEW

File: docs/release-notes.md [hunk 1]
```diff
@@ -1,3 +1,21 @@
...
+### Added
+* `Response.delete_cookie` now accepts the same parameters as `Response.set_cookie` [#1228](https://github.com/encode/starlette/pull/1228).
+* Update the `Jinja2Templates` constructor to allow `PathLike` [#1292](https://github.com/encode/starlette/pull/1292).
+
+### Fixed
+* Fix BadSignature exception handling in SessionMiddleware [#1264](https://github.com/encode/starlette/pull/1264).
+* Change `HTTPConnection.__getitem__` return type from `str` to `typing.Any` [#1118](https://github.com/encode/starlette/pull/1118).
+* Change `ImmutableMultiDict.getlist` return type from `typing.List[str]` to `typing.List[typing.Any]` [#1235](https://github.com/encode/starlette/pull/1235).
+* Handle `OSError` exceptions on `StaticFiles` [#1220](https://github.com/encode/starlette/pull/1220).
+* Fix `StaticFiles` 404.html in HTML mode [#1314](https://github.com/encode/starlette/pull/1314).
```
Comment (Kludex) [inline docs/release-notes.md:14]:
```suggestion
* Fix `StaticFiles` 404.html in HTML mode [#1314](https://github.com/encode/starlette/pull/1314).
* Prevent anyio.ExceptionGroup in error views under a BaseHTTPMiddleware [#1262](https://github.com/encode/starlette/pull/1262).
```

**Comment by havardthom** [comment]
Will https://github.com/encode/starlette/pull/1318 be in 0.17.0?

**Comment by Kludex** [comment]
@havardthom I can't reproduce #1284 against master after #1262 merge.

**Review by florimondmanca — APPROVED** [review]
Reviewed the changes and the compare log, looking good to me.

**Comment by Kludex** [comment]
Thanks @florimondmanca ! :sunglasses: 

Just for the record, documentation and CI changes were not included on the changelog - in `httpx` that's written in the `CONTRIBUTING.md`, but we don't have contribution guidelines on `starlette` yet.

**Comment by florimondmanca** [comment]
@Kludex Yup I think that's a good practice in general — only include changes that matter to code users. :+1:

**Pull request merged**